amy/workshops/templatetags/training_progress.py

```diff
@@ -36,11 +36,17 @@ def progress_description(progress):
     else:
         involvement = ""
 
+    day = (
+        progress.date.strftime("%A %d %B %Y")
+        if progress.date
+        else progress.created_at.strftime("%A %d %B %Y at %H:%M")
+    )
+
     text = "{state} {type}{involvement}<br />on {day}.{notes}".format(
```

Comment:
I found this not very readable in your screenshot. Does the type have to be repeated (it's also displayed on the label)?

Reply:
Would you like to add tests for this template tag?

Reply:
Thanks for the readability comment, I agree the wording becomes quite weird in this case - I will ask for some feedback from the ITT on this and see what would work best for them.

I completely overlooked testing of this tag, but I will add some tests either here or in a follow-up PR depending on timing.